src/containers/Application/ApplicationPageWrapper.tsx

```diff
@@ -111,4 +167,84 @@ function processRedirect(appState: any): void {
   }
 }
 
+interface buildProgressInApplicationProps {
+  elementsState: ApplicationElementStates | undefined
+  responses: ResponsesFullByCode | undefined
+  templateSections: TemplateSectionPayload[]
+  isLinear: boolean
+  currentSection: number
+  currentPage: number
+}
+
+function buildProgressInApplication({
+  elementsState,
+  responses,
+  templateSections,
+  isLinear,
+  currentSection,
+  currentPage,
+}: buildProgressInApplicationProps): ProgressInApplication {
+  if (!elementsState || !responses) return []
+  let sectionsStructure: ProgressInApplication = templateSections.map((section) => {
+    // Create an array with all pages in each section
+    const pageNumbers = Array.from(Array(section.totalPages).keys(), (n) => n + 1)
+    const isCurrentPage = (page: number) => section.index === currentSection && page === currentPage
+    const isPreviousPage = (sectionIndex: number, page: number) =>
+      sectionIndex <= currentSection || page <= currentPage
+
+    const pages = pageNumbers.map((pageNumber) => ({
+      pageName: `Page ${pageNumber}`,
+      canNavigate: isLinear ? isPreviousPage(section.index, pageNumber) : true,
+      isActive: isCurrentPage(pageNumber),
+      status: isCurrentPage(pageNumber)
```

Comment:
I still don't quite understand the need for condition like this here.

If you forget about `strict` mode setting in validationPage -> it just return valid, invalid or incomplete.

Then the logic seem quite simple to me:
```
getStatue = () => {
draftPageStatus = validatePage
if(isLinear || strict) return draftPageStatus === valid : valid ? not valid

return draftPageStatus
}
```

And for canNavigate
```
canNavigate = true
...
let sectionsStructure: ProgressInApplication = templateSections.map((section) => {
...
const pages = pageNumbers.map((pageNumber) => {
...
status = getPageStatus()
returnObject = {
...
status,
canNavigate
...
}
...
canNavigate = status !== not valid
...
return returnObject
}

```